rollup/pipeline/pipeline.go

```diff
@@ -165,9 +165,11 @@ func (p *Pipeline) TryPushTxn(tx *types.Transaction) (*Result, error) {
 	}
 }
 
-func (p *Pipeline) Kill() {
+// Release releases all resources related to the pipeline
+func (p *Pipeline) Release() {
 	if p.txnQueue != nil {
 		close(p.txnQueue)
+		p.txnQueue = nil
 	}
 
 	select {
```

Comment:
not understand when doing Release, why these channels will always return values.
e.g. case 1: `p.txnQueue != nil`, case 2: `p.txnQueue == nil`.

Reply:
> why these channels will always return values

they won't neccessarily return values but they will be closed by the stages that own them.

Reply:
`Release()` waits for the 2 channels to return (anything) after closing `p.txnQueue`, I found another place to close it in advance but it looks like it will never be triggered because w.onTxFailingInPipeline always returns false in the current implementation.